### PR TITLE
Add common CQRS infrastructure

### LINKED ISSUE #278: Add common CQRS infrastructure
### Description
Add common infrastructure for CQRS and async communication

### Requirements

- project must be runnable
- mediatr must be registered to the middleware with loading types using reflection

### Deliverables and requirements

- `IQuery` - interface for all query requests
- `ICommand` - interface for all command requests
- `IEvent` - interface for all events
- `EventBroker` - broker for rabbit mq
- `RabbitMqConnectionProvider` - connection provider of rabbit mq
- `EventSubscriber` - event subscriber

### Reference

Models : 
- https://github.com/sulton-max/Backend.ILA.Api-Comms.Learning/tree/main/src/N2.Microservices.Common.Core/Models

Registering mediator  : 
- https://github.com/sulton-max/Backend.ILA.Api-Comms.Learning/blob/main/src/N2.Microservices.Services.BankingService.Api/Configurations/HostConfigurations.Extensions.cs

### PR DESCRIPTION
**Pull request by rashidovasadbek**
Closes #278

### CODE REVIEW

File: src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/EventBus/IEvenBusBroker.cs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    /// <summary>
+    /// Subscribes a handler to an event, allowing it to handle the event when published.
+    /// </summary>
+    /// <typeparam name="TEvent">The type of the event to subscribe to.</typeparam>
+    /// <typeparam name="TEventHandler">The type of the event handler that will handle the event.</typeparam>
+    /// <returns>A <see cref="ValueTask"/> representing the asynchronous operation.</returns>
+    /// <remarks>
+    /// The event handler must implement the <see cref="IEventHandler{TEvent}"/> interface for proper event handling.
+    /// </remarks>
+    ValueTask SubscribeAsync<TEvent, TEventHandler>() where TEvent : IEvent where TEventHandler : IEventHandler<TEvent>;
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/EventBus/IEvenBusBroker.cs:38]: Remove subscriber method, create `EventSubscriber` generic service

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by Ilxom-22 — APPROVED** [review]

File: src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/EventBus/Brokers/IEventSubscriber.cs [hunk 2]
```diff
@@ -0,0 +1,8 @@
+﻿namespace AirBnB.Application.Common.EventBus.Brokers;
+
+public interface IEventSubscriber
+{
+    ValueTask StartAsync(CancellationToken token);
+
+    ValueTask StopAsync(CancellationToken token);
+}
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/EventBus/Brokers/IEventSubscriber.cs:8]: Add comments

File: src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/EventBus/Brokers/IEvenBusBroker.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+using AirBnB.Domain.Events;
+
+namespace AirBnB.Application.Common.EventBus.Brokers;
+
+/// <summary>
+/// Represents a generic interface for an event bus broker, which facilitates the publishing and subscribing to events.
+/// </summary>
+public interface IEvenBusBroker
+{
+    ValueTask PublishAsync<TEvent>(TEvent @event, string exchange, string routingKey, CancellationToken cancellationToken) where TEvent : Event;
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/EventBus/Brokers/IEvenBusBroker.cs:11]: Add `PublishLocalAsync` which should send events to MediatR

File: src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/EventBus/Brokers/IRabbitMqConnectionProvider.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+/// </summary>
+public interface IRabbitMqConnectionProvider
+{
+    /// <summary>
+    /// Asynchronously creates a RabbitMQ channel for communication.
+    /// </summary>
+    /// <returns>A <see cref="ValueTask{TResult}"/> representing the asynchronous operation. The result is the created RabbitMQ channel.</returns>
+    /// <remarks>
+    /// Creating a channel is an asynchronous operation that allows for the establishment of a communication channel with RabbitMQ for sending and receiving messages.
+    /// </remarks>
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Application/Common/EventBus/Brokers/IRabbitMqConnectionProvider.cs:16]: Remove this

File: src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Settings/NotificationSubscriberSettings.cs [hunk 5]
```diff
@@ -0,0 +1,6 @@
+﻿namespace AirBnB.Infrastructure.Common.Settings;
+
+public class NotificationSubscriberSettings
+{
+    
+}
```
Comment (sulton-max) [inline src/Training.AirBnb.Clone.Backend/AirBnB.Infrastructure/Common/Settings/NotificationSubscriberSettings.cs:6]: Do we need this in this task ?

**Review by sulton-max — CHANGES_REQUESTED** [review]

**Review by sulton-max — APPROVED** [review]
Good job

**Review by botirjonabdumannonovv — APPROVED** [review]

**Review by Ilxom-22 — APPROVED** [review]

**Pull request merged**